go/consensus/tendermint/apps/registry/registry.go

```diff
@@ -121,6 +122,17 @@ func (app *registryApplication) ExecuteTx(ctx *api.Context, tx *transaction.Tran
 			return err
 		}
 		return nil
+
+	case registry.MethodProveFreshness:
+		var blob [32]byte
+		if err := cbor.Unmarshal(tx.Body, &blob); err != nil {
+			return err
```

Comment:
To avoid this being consensus breaking, this must return `registry.ErrInvalidArgument` in case freshness proofs are not supported. Otherwise the resulting error module/code may be different, causing a consensus failure.

Looking back all of these above should actually return `registry.ErrInvalidArgument` during deserialization failures, but we can't change this now as it would be consensus-breaking.

Reply:
Agree with both statements.